persistence-api/src/main/java/io/stargate/db/schema/SchemaHash.java

```diff
@@ -0,0 +1,68 @@
+package io.stargate.db.schema;
+
+import java.util.Collection;
+import java.util.Objects;
+
+/** An interface for calculating deterministic hash codes for schema objects. */
+public interface SchemaHash {
```

Comment:
nit: can we change the name to something like `SchemaHashable`? Otherwise it suggests that a schema entity "is a" schema hash, whereas it's something that _produces_ a schema hash.